qemu/tests/fio_perf.py

```diff
@@ -0,0 +1,224 @@
+import logging
+import os
+import commands
+import threading
+import re
+import time
+from autotest.client import utils
+from autotest.client.shared import error
+from virttest import utils_misc, utils_test
+
+
+def format_result(result, base="12", fbase="2"):
+    """
+    Format the result to a fixed length string.
+
+    :param result: result need to convert
+    :param base: the length of converted string
+    :param fbase: the decimal digit for float
+    """
+    if isinstance(result, str):
+        value = "%" + base + "s"
+    elif isinstance(result, int):
+        value = "%" + base + "d"
+    elif isinstance(result, float):
+        value = "%" + base + "." + fbase + "f"
+    return value % result
+
+
+def check_status(session, timeout, num):
+    """
+    Output disk info including disk status
+
+    :param session: VM session
+    :param timeout: Timeout in seconds
+    :param num: The number of data disks
+    """
+
+    disk_status_cmd = "echo list disk > cmd && echo exit >>"
+    disk_status_cmd += " cmd && diskpart /s cmd"
+    disks = []
+    start_time = time.time()
+    while time.time() - start_time < timeout:
```

Comment:
it looks performance of below code looks better than your code.
end_time = time.time() + timeout
while time.time() < end_timeout:
      do something